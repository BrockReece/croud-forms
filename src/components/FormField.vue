<template>
    <div class="ui field" :class="inputClasses">
        <label>{{ field.field_name }}</label>
        <div class="" v-for="(val, $key) in validation" v-if="validation.$dirty && !$key.startsWith('$') && !val" :key="$key">{{ validation.$params[$key] | displayError }}</div>
        <component :is="view" :field="field" :placeholder="field.field_name" :transparent="transparent" :read-only="readOnly" :model="model" @set-value="(val) => { $emit('set-value', val) }" :validation="validation"></component>
    </div>
</template>

<script>
    import FormMixin from '../mixins/Form'

    export default {
        mixins: [FormMixin],
    }
</script>
