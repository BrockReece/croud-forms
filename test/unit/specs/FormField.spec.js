import { mount } from 'vue-test-utils'
import FormField from '@/components/FormField'

const fieldTypes = ['text', 'number']

describe('FormField component', () => {
    describe('should render correct field type', () => {
        fieldTypes.forEach((type) => {
            it(type, () => {
                const vm = mount(FormField, {
                    propsData: {
                        field: {
                            field_name: 'First name',
                            field_type: type,
                        },
                        model: 'test',
                    },
                })
                expect(vm.html()).toMatchSnapshot()
            })
        })
    })

    it('should render a transparent input', () => {
        const vm = mount(FormField, {
            propsData: {
                field: {
                    field_name: 'First name',
                    field_type: 'text',
                },
                model: 'test',
                transparent: true,
            },
        })
        expect(vm.html()).toMatchSnapshot()
    })
})
