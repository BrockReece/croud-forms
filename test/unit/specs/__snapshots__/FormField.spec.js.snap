// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component should render a transparent input 1`] = `
<div class="ui field"><label>First name</label>
    <div class="ui fluid input transparent"><input type="text" placeholder="First name"></div>
</div>
`;

exports[`FormField component should render correct field type checkbox 1`] = `
<div class="ui field"><label>First name</label>
    <div class="ui input">
        <semantic-checkbox type="checkbox"></semantic-checkbox>
    </div>
</div>
`;

exports[`FormField component should render correct field type number 1`] = `
<div class="ui field"><label>First name</label>
    <div class="ui input"><input type="number" placeholder="First name"></div>
</div>
`;

exports[`FormField component should render correct field type text 1`] = `
<div class="ui field"><label>First name</label>
    <div class="ui fluid input"><input type="text" placeholder="First name"></div>
</div>
`;
